### PR TITLE
MediaPicker: New source to load media using product ID

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #12540

## Description

Adds a new case to `MediaPickingSource` to select media attached to a given product ID.

The new `productMedia` source option is not used yet. We will use it to load specific product images for the Blaze Edit Ad screen in a future PR. 

Changes
- New case in `MediaPickingSource` to use product ID to retrieve media 
- Add a way to inject media source options in `MediaSourceActionSheet`
- Pass product ID to `WordPressMediaLibraryPicker` related code and retrieve media using the given product ID. 
- Added internal release notes for beta testing. I will remove it when I raise the final PR to use the new source option.
- Unit tests.

## Steps to reproduce

1. Create a JN site with Woo and Jetpack. 
2. Add media to your from the wp-admin page.
3. Login into the store using WPORG site credentials.
4. Navigate to the `Products` tab -> Tap `+` button from top right to start product creation flow.
5. In the product creation form, tap on the "Add a product image" button
6. Validate that the media picker options are presented as before.
7. Choose "WordPress Media Library"
8. Ensure that all the media images from your store are loaded correctly.
9. Log out of the mobile app.
10. Login into the mobile app using your WPCOM account.
11. Repeat steps 4 to 8.

## Testing information

- The new `productMedia(productID: Int64)`media source case is not used anywhere yet. 
- We need to test that the Media Picker implementation works as before.

## Screenshots

---
- [x] I have considered adding unit tests for this change. If I decided not to add them, I have provided a brief explanation below (optional):
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
- [ ] This PR includes refactoring; smoke testing of the entire section is needed.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/View Modifiers/View+MediaSourceActionSheet.swift [hunk 1]
```diff
@@ -3,50 +3,71 @@ import SwiftUI
...
         )
         static let photoLibrary = NSLocalizedString(
-            "Choose from device",
+            "mediaSourceActionSheet.photoLibrary",
+            value: "Choose from device",
             comment: "Menu option for selecting media from the device's photo library."
         )
+        static let productMedia = NSLocalizedString(
+            "mediaSourceActionSheet.productMedia",
+            value: "Choose from product media",
```
Comment (itsmeichigo) [inline WooCommerce/Classes/View Modifiers/View+MediaSourceActionSheet.swift:65]: Nit: Would it be clearer to use "Choose an existing product photo"?

**Review by itsmeichigo — APPROVED** [review]
LGTM :shipit:

File: WooCommerce/Classes/View Modifiers/View+MediaSourceActionSheet.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/View Modifiers/View+MediaSourceActionSheet.swift:65]: Good suggestion. I changed it [here](https://github.com/woocommerce/woocommerce-ios/pull/13540/commits/ce00414178858d1c6f9823c16b10eb02c8a3b503) in this PR. https://github.com/woocommerce/woocommerce-ios/pull/13540

**Pull request merged**